src/main/java/io/fabric8/jenkins/openshiftsync/BuildConfigToJobMapper.java

```diff
@@ -116,88 +133,103 @@ public static FlowDefinition mapBuildConfigToFlow(BuildConfig bc) throws IOExcep
 	 * @return true if the BuildConfig was changed
 	 */
 	public static boolean updateBuildConfigFromJob(WorkflowJob job, BuildConfig buildConfig) {
-		NamespaceName namespaceName = NamespaceName.create(buildConfig);
-		JenkinsPipelineBuildStrategy jenkinsPipelineStrategy = null;
-		BuildConfigSpec spec = buildConfig.getSpec();
-		if (spec != null) {
-			BuildStrategy strategy = spec.getStrategy();
-			if (strategy != null) {
-				jenkinsPipelineStrategy = strategy.getJenkinsPipelineStrategy();
-			}
-		}
+    NamespaceName namespaceName = NamespaceName.create(buildConfig);
+    JenkinsPipelineBuildStrategy jenkinsPipelineStrategy = null;
+    BuildConfigSpec spec = buildConfig.getSpec();
+    if (spec != null) {
+      BuildStrategy strategy = spec.getStrategy();
+      if (strategy != null) {
+        jenkinsPipelineStrategy = strategy.getJenkinsPipelineStrategy();
+      }
+    }
 
-		if (jenkinsPipelineStrategy == null) {
-			LOGGER.warning("No jenkinsPipelineStrategy available in the BuildConfig " + namespaceName);
-			return false;
-		}
+    if (jenkinsPipelineStrategy == null) {
+      LOGGER.warning("No jenkinsPipelineStrategy available in the BuildConfig " + namespaceName);
+      return false;
+    }
 
-		FlowDefinition definition = job.getDefinition();
-		if (definition instanceof CpsScmFlowDefinition) {
-			CpsScmFlowDefinition cpsScmFlowDefinition = (CpsScmFlowDefinition) definition;
-			String scriptPath = cpsScmFlowDefinition.getScriptPath();
-			if (scriptPath != null && scriptPath.trim().length() > 0) {
-				boolean rc = false;
-				BuildSource source = getOrCreateBuildSource(spec);
-				String bcContextDir = source.getContextDir();
-				if (StringUtils.isNotBlank(bcContextDir) && scriptPath.startsWith(bcContextDir)) {
-					scriptPath = scriptPath.replaceFirst("^" + bcContextDir + "/?", "");
-				}
+    LOGGER.info("Updating BuildConfig From Job "+namespaceName);
 
-				if (!scriptPath.equals(jenkinsPipelineStrategy.getJenkinsfilePath())) {
-					LOGGER.log(Level.FINE,
-							"updating bc " + namespaceName + " jenkinsfile path to " + scriptPath + " from ");
-					rc = true;
-					jenkinsPipelineStrategy.setJenkinsfilePath(scriptPath);
-				}
+    FlowDefinition definition = job.getDefinition();
+    if (definition instanceof CpsScmFlowDefinition) {
```

Comment:
null check ?

Reply:
instanceof will be false if `definition` is `null`.